presto-native-execution/presto_cpp/main/functions/remote/PrestoToVeloxRemoteFunctionExpr.cpp

```diff
@@ -0,0 +1,195 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/functions/remote/PrestoToVeloxRemoteFunctionExpr.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/functions/remote/client/RestRemoteClient.h"
+#include "presto_cpp/main/functions/remote/client/VeloxRemoteFunction.h"
+
+using facebook::velox::functions::remote::PageFormat;
+
+namespace facebook::presto::functions::remote::rest {
+namespace {
+// Returns the serialization/deserialization format used by the remote function
+// server. The format is determined by the system configuration value
+// "remoteFunctionServerSerde". Supported formats:
+//   - "presto_page": Uses Presto page format.
+//   - "spark_unsafe_row": Uses Spark unsafe row format.
+// @return PageFormat enum value corresponding to the configured serde format.
+// @throws VeloxException if the configured format is unknown.
+PageFormat getSerde() {
+  static const auto serdeFormat =
+      SystemConfig::instance()->remoteFunctionServerSerde();
+  if (serdeFormat == "presto_page") {
+    return PageFormat::PRESTO_PAGE;
+  } else if (serdeFormat == "spark_unsafe_row") {
+    return PageFormat::SPARK_UNSAFE_ROW;
+  } else {
+    VELOX_FAIL(
+        "Unknown serde name for remote function server: '{}'", serdeFormat);
+  }
+}
+
+// Extracts the schema name from a fully qualified Presto function identifier.
+// The function identifier is expected in the format
+// "namespace.schema.function;TYPE;TYPE". This function returns the substring
+// between the first and second dots in the function name.If the schema cannot
+// be determined, "default" is returned.
+// @param functionId The fully qualified Presto function identifier.
+// @return The extracted schema name or "default".
+std::string getSchemaName(const protocol::SqlFunctionId& functionId) {
+  // Example: "namespace.schema.function;TYPE;TYPE".
+  const auto nameEnd = functionId.find(';');
+  std::string functionName = (nameEnd != std::string::npos)
+      ? functionId.substr(0, nameEnd)
+      : functionId;
+
+  const auto firstDot = functionName.find('.');
+  const auto secondDot = functionName.find('.', firstDot + 1);
+  if (firstDot != std::string::npos && secondDot != std::string::npos) {
+    return functionName.substr(firstDot + 1, secondDot - firstDot - 1);
+  }
+
+  return "default";
+}
+
+// Extracts the function name from a fully qualified function identifier string.
+// The input is expected to be in the format "namespace.schema.function", and
+// this function returns the substring after the last dot. If there is no dot,
+// the entire input string is returned.
+// @param input The fully qualified function identifier.
+// @return The extracted function name.
+std::string getQualifiedObjectName(const std::string& input) {
```

Comment:
Can be removed and function name can be extracted with `getFunctionNameParts` in caller?